airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -410,12 +403,32 @@ def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Po
             self.volumes +  # type: ignore
             self.volume_mounts  # type: ignore
         )
+        return pod
 
-        self.pod = pod
+    def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Pod, Optional[str]]:
+        """
+        Creates a new pod and monitors for duration of task
+
+        :param labels: labels used to track pod
+        :param launcher: pod launcher that will manage launching and monitoring pods
+        :return:
+        """
+        if not (self.full_pod_spec or self.pod_template_file):
+            # Add Airflow Version to the label
+            # And a label to identify that pod is launched by KubernetesPodOperator
+            self.labels.update(
+                {
+                    'airflow_version': airflow_version.replace('+', '-'),
+                    'kubernetes_pod_operator': 'True',
+                }
+            )
+            self.labels.update(labels)
+            self.pod.metadata.labels = self.labels
+        pod = self.pod
```

Comment:
Should we remove tthis variable?  This is a bit confusing because in one function we refer to the same object in two ways - `pod`, `self.pod`.